eng/scripts/test-matrix/test-matrix-functions.ps1

```diff
@@ -0,0 +1,317 @@
+Set-StrictMode -Version "4.0"
+
+function CreateDisplayName([string]$parameter, [hashtable]$displayNames) {
+    $name = $parameter
+
+    if ($displayNames[$parameter]) {
+        $name = $displayNames[$parameter]
+    }
+
+    return $name -replace "[\-\._]", ""
+}
+
+function GenerateMatrix([HashTable]$config, [string]$selectFromMatrixType) {
+    if ($selectFromMatrixType -eq "sparse") {
+        [Array]$dimensions = GetMatrixDimensions $config.matrix
+        $size = $dimensions[0]
+        [Array]$matrix = GenerateSparseMatrix $config.matrix $config.displayNames $size
+    } elseif ($selectFromMatrixType -eq "all") {
+        [Array]$matrix, $_ = GenerateFullMatrix $config.matrix $config.displayNames
+    } else {
+        throw "Matrix generator not implemented for selectFromMatrix: $($platform.selectFromMatrix)"
+    }
+
+    if ($config["exclude"]) {
+        [Array]$matrix = ProcessExcludes $matrix $config.exclude
+    }
+    if ($config["include"]) {
+        [Array]$matrix = ProcessIncludes $matrix $config.include $config.displayNames
+    }
+
+    return $matrix | Sort-Object -Property name
+}
+
+function ProcessExcludes([Array]$matrix, [Array]$excludes) {
+    $deleteKey = "%DELETE%"
+    $exclusionMatrix = @()
+
+    foreach ($exclusion in $excludes) {
+        $converted = ConvertToMatrixArrayFormat $exclusion
+        $full, $_ = GenerateFullMatrix $converted
```

Comment:
What is `$_` doing in this context? Is it just throwing away the second parameter?

Reply:
Yes. Originally the function returned an object with two keys, but I thought that made writing other code more error prone, so I switched it to return positional values. But not all callers care about $dimensions.